project/api/models.py

```diff
@@ -87,9 +90,18 @@ def clean(self, *args, **kwargs):
         if self.start_date > self.end_date:
             raise ValidationError("Start date cannot be later than end date")
         if loads(self.geo.json)["type"] != "Polygon" and loads(self.geo.json)["type"] != "MultiPolygon":
-            raise ValidationError("Only Polygon and MultiPolygon objects are acceptable geometry types")
+            raise ValidationError(
+                "Only Polygon and MultiPolygon objects are acceptable geometry types")
         super(Territory, self).clean(*args, **kwargs)
 
+        # This date check is inculsive.
+        if Territory.objects.filter(start_date__range=(self.start_date, self.end_date), nation=self.nation).exists() or Territory.objects.filter(end_date__range=(self.start_date, self.end_date), nation=self.nation).exists():
+            raise ValidationError(
+                "Another territory of this nation exists during this timeframe.")
+        if Territory.objects.filter(start_date__lte=self.end_date, end_date__gte=self.end_date).exists() and Territory.objects.filter(start_date__lte=self.start_date, end_date__gte=self.start_date).exists():
```

Comment:
There is a more optimized way. The 2 filters can be simplified in:
`Territory.objects.filter(start_date__lte=self.end_date, end_date__gte=self.start_date).exists()`

Proof:
```
                  |s1----------e1| 
|s2----------e2|
```
`[s1,e1]` and `[s2,e2]` overlap iif `(s1<=e2 and s2<=e1)`